### ISSUE TITLE
Incorrect error message under Firebase/"Delete User Reference"

### ISSUE THREAD

**Issue by AdrKacz** [FlutterFlow/flutterflow-issues#5145]
### Can we access your project?

- [X] I give permission for members of the FlutterFlow team to access and test my project for the sole purpose of investigating this issue.

### Current Behavior

After deleting a collection referenced in the "Delete User Reference" function, selecting "Update" triggers the error: *"Unknown error. Please contact [support@flutterflow.io](mailto:support@flutterflow.io)."*  

While this appears concerning, it’s not a major issue. Upon checking in Google Cloud, the function updates with the expected value, even though the FlutterFlow UI continues to display "Out of Date" or "Error."

### Expected Behavior

I’d expect FlutterFlow to fetch the function source directly from Google Cloud.  

Currently, it seems FlutterFlow infers the source from previous successful deployments. However, if a deployment fails, subsequent predictions become inaccurate. Instead, it should fetch the actual source code and compare it to the expected value.  

If this is already the case, I’d expect FlutterFlow to perform accurate string comparisons, as I’ve manually confirmed that both the expected and actual sources are identical.

### Steps to Reproduce

1. Create some collections
2. Create some rules so your collections are deleted with your user
3. Delete one of these collection (not the users one)
4. Change environment (the error may appear also without changing the environment, however I didn't verify)

### Reproducible from Blank

- [X] The steps to reproduce above start from a blank project.

### Bug Report Code (Required)

Skip

### Visual documentation

Skip

### Environment

```markdown
- FlutterFlow version: 5.0.18
- Platform: Desktop
- Browser name and version: _
- Operating system and version affected: MacOS
```

### Additional Information

Skip

**Comment by paulperez-dev** [comment]
Hi @AdrKacz,

Thanks for the details so far! Could you let me know how to trigger the error message you mentioned after step 4? If possible, could you upload a video showing the steps to reproduce the issue and include the bug report code as an attachment? That would really help us understand the problem better.

Appreciate your time and effort—looking forward to your response!

**Comment by AdrKacz** [comment]
What steps did you take to reproduce the error, and what was the result? The instructions I provided should be sufficient, but I’d need more details to revise them if needed.

I’ve noticed that every Firebase deployment returns an error, even when the deployment succeeds. It seems FlutterFlow is still verifying a deleted collection, which triggers an error during the verification step (after the deployment request has been issued and completed).

**Comment by AdrKacz** [comment]
This issue have been solved in the latest version of FlutterFlow (5.0.21), thanks for your help.
